cli/generator.go

```diff
@@ -1153,6 +1153,28 @@ func formatKebab[T ~string](name T) string {
 	return kebabName
 }
 
+// parseDurationExtended parses a duration string with optional "w" (weeks) and "d" (days) suffixes or standard time.Duration formats.
+// If the input uses "w" or "d", it is converted to time.Duration; otherwise, it falls back to parsing via time.ParseDuration.
+// Returns the equivalent time.Duration and any error encountered during conversion.
```

Comment:
This might not work for combinations of units such as "2d12h" or "1w2d". This functionality used to be provided by a prometheus util library but I removed it the other day to remove the dependency on prometheus. Maybe it will be easier to copy the logic from that library?